### ISSUE TITLE
Beetle Adventure Racing Fog stutters with bands of non-fog area

### ISSUE THREAD

**Issue by matthewharvey** [gonetz/GLideN64#1541]
Linux with an Intel Ivybridge card. I tried using both the development version and whatever my distro uses for the Mesa drivers.

Bisected to ebcf46e5adc0fe8717143a8e31949bc09247a149

I tried to attach a picture, but was unable. There are bands of the screen that are not covered in fog. On first sight it kind of looks like a driver issue, but it was introduced by a commit to GLideN64.

**Comment by loganmc10** [comment]
Unfortunately I don't have a machine with Intel graphics + Linux that supports the fog so I can't test this.

Can you try modifying this line:
https://github.com/gonetz/GLideN64/blob/master/src/Graphics/OpenGLContext/opengl_ContextImpl.cpp#L306

And remove the ```&& m_glInfo.renderer != Renderer::Intel``` bit, that will help me narrow down where the problems lies

**Comment by matthewharvey** [comment]
I just did a race with that change. It was happening still, but much, much less often. As you approach certain objects and the fog lifts, there is an in-between place where some of them will occasionally have those bands, and when they get the bands they have them in "full force" just like in the picture I posted. It's often the green hills, but that might be because it's much more noticeable on large, dark objects.

I can keep providing information, and will also be checking out earlier versions of the Mesa driver, and probably will try out a debug build of the driver just for fun.

**Comment by loganmc10** [comment]
Yeah I tested both implementations of ColorBufferReader on my laptop with an Nvidia card, the fog looked normal (tested Mount Mayhem). No GL errors reported.

**Comment by gonetz** [comment]
Beetle Adventure Racing uses depth buffer based fog. It is implemented in shaders and should work only if OpenGL supports image textures.

**Comment by matthewharvey** [comment]
My driver has GL_ARB_shader_image_load_store, and GL_ARB_texture_buffer_object.

Which extension is needed for the fog? Also, did the requirements change recently, because up until a few months ago the fog was working well on my machine?

**Comment by gonetz** [comment]
Test for image textures support:
```
imageTextures = ((numericVersion >= 43) || (Utils::isExtensionSupported(*this, "GL_ARB_shader_image_load_store") &&
Utils::isExtensionSupported(*this, "GL_ARB_compute_shader"))) && IS_GL_FUNCTION_VALID(glBindImageTexture);

```

That is either 
GL version 4.3 or above 
or 
GL_ARB_shader_image_load_store and GL_ARB_compute_shader extensions and glBindImageTexture  function found.

**Comment by matthewharvey** [comment]
I have compute shaders as well. I'm pretty sure image textures should be working for me.

Anyway, this was fixed by 032da853572be5074ffdc2c0c4f03dc20d4de431 which is the fix for defect 1594 which was complaining about line flickers which sounds similar to what I was seeing. 
I'm closing this, as it is no longer an issue for me.
